### ISSUE TITLE
Tensorboard epoch summaries dependent on divisibility by update_freq

### ISSUE THREAD

**Issue by atyshka** [tensorflow/tensorflow#49109]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 20.04
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): 2.4.1
- Python version: 3.8
- CUDA/cuDNN version: 11.0/8.0
- GPU model and memory: 2x RTX A6000

**Describe the current behavior**
I am training on a large dataset, and for tensorboard I would like to evaluate certain metrics such as loss at a higher frequency, while other metrics such as test images I would like to evaluate more infrequently, such as at the end of a batch. To use a more frequent loss, I set update_freq on the keras Tensorboard callback to be a reasonably small number so as to get frequent updates. Then to log my images, I setup a lambda callback on epoch end. However, I was confused that the images were never being written. I eventually discovered that the on_epoch_end lambda callback will not be able to log anything via tf.summary unless the dataset length is divisible by the update_freq. For example, if my update_freq was 5, and my dataset was 10 steps long, the epoch callbacks would be able to write, but if the dataset was 11 steps long, the callback would not fire since 11 is not a multiple of 5. Since my lambda callback and tensorboard callbacks are separate objects, it is not intuitive that the update_freq of the tensorboard callback would affect the lambda callback. Of course the simple solution is to truncate my dataset to a number divisible by update_freq, but that is both non-intuitive and discards valuable data. 

**Describe the expected behavior**
I would expect that lambda callbacks should fire on_epoch_end events regardless of update_freq, or at the very least that this behavior be documented to save others time if they encounter this issue.

**Standalone code to reproduce the issue**

```
tboard_callback = tf.keras.callbacks.TensorBoard(log_dir = logs, update_freq=5)
file_writer = tf.summary.create_file_writer('logs/images')
def evaluate_images(epoch, logs):
    print("Writing images")
    with file_writer.as_default():
        print(tf.summary.image("Prediction", tf.zeros([1, 100, 100, 1])))
        # Prints false if images are not written because epoch length not divisible by update_freq
model.fit(dataset, epochs=1, callbacks = [tboard_callback, tf.keras.callbacks.LambdaCallback(on_epoch_end=evaluate_images)])
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by tilakrayal** [comment]
@atyshka ,

In order to reproduce the issue reported here, could you please provide the complete code and the dataset you are using. Thanks!
